ddtrace/internal/telemetry/data.py

```diff
@@ -0,0 +1,91 @@
+import platform
+import sys
+from typing import Dict
+from typing import Tuple
+
+from ddtrace.internal.compat import PY3
+from ddtrace.internal.runtime.container import get_container_info
+from ddtrace.internal.utils.cache import cached
+
+from ...version import get_version
+from ..hostname import get_hostname
+
+
+def _format_version_info(vi):
+    # type: (sys._version_info) -> str
+    """
+    Converts sys.version_info into a string with the format x.x.x
+    """
+    return "%d.%d.%d" % (vi.major, vi.minor, vi.micro)
+
+
+def _get_container_id():
+    # type: () -> str
+    """
+    Helper function private to this module
+    Get ID from docker container
+    """
+    container_info = get_container_info()
+    if container_info:
+        return container_info.container_id or ""
+    return ""
+
+
+def _get_os_version():
+    # type: () -> str
+    """
+    Helper function private to this module
+    Returns the os version for applications running on Unix, Mac or Windows 32-bit
+    """
+    mver, _, _ = platform.mac_ver()
+    _, wver, _, _ = platform.win32_ver()
+    _, lver = platform.libc_ver()
+
+    return mver or wver or lver or ""
+
+
+@cached()
+def _get_application(key):
+    # type: (Tuple[str, str, str]) -> Dict
+    """
+    This helper packs and unpacks get_application arguments to support caching.
+    Cached() annotation only supports functions with one argument
+    """
+    service, version, env = key
+
+    return {
+        "service_name": service or "unnamed_python_service",
+        "service_version": version or "",
+        "env": env or "",
+        "language_name": "python",
+        "language_version": _format_version_info(sys.version_info),
+        "tracer_version": get_version(),
+        "runtime_name": platform.python_implementation(),
+        "runtime_version": _format_version_info(sys.implementation.version) if PY3 else "",
+    }
+
+
+def get_application(service, version, env):
+    """Creates a dictionary to store application data using ddtrace configurations and the System-Specific module"""
+    # Application Dict is cached when service, version and env configurations change during runtime
```

Comment:
Can we adjust this comment to more clearly state that we have to call `get_application()` at runtime in case `service`, `version`, or `env` get changed after startup/during runtime. Because we have to call it at runtime we cache the results to reduce overhead.